### ISSUE TITLE
Provide feedback to user when there is missing internet connection

### ISSUE THREAD

**Issue by russnes** [dhis2/dhis2-android-trackercapture#43]
In many different scenarios a connection to the server may fail or something in the app may not work as expected if there is a missing or unstable internet connection. 

These include:
- Logging in
- Loading meta data (initial)
- Updating meta data
- Loading data values
- Send data values

As of now, there isn't really any feedback given to the user that makes much sense. 
For logging in, a message saying "No address associated with the www. " is shown to the user, which can be confusing.
For the other scenarios no real feedback is provided.
For meta data loading it will retry if it fails, but it will not say why.
For sending data values, a more detailed error message is always displayed when something other than the connection fails, but when the connection fails, nothing happens at all.

So basically, make the feedback in these scenarios better.
